spring-kitchensink-basic/src/main/java/org/jboss/as/quickstarts/kitchensink/spring/basic/controller/MemberController.java

```diff
@@ -51,7 +53,14 @@ public String registerNewMember(@Valid @ModelAttribute("newMember") Member newMe
                 return "redirect:/";
             } catch (UnexpectedRollbackException e) {
                 model.addAttribute("members", memberDao.findAllOrderedByName());
-                model.addAttribute("error", e.getCause().getCause());
+                // Check the uniqueness of the email address
+                if (emailAlreadyExists(newMember.getEmail())) {
+                    System.out.println("****** ConstraintViolationException: Duplicate email");
```

Comment:
I suggest we use a logger here since we are trying to "teach" users the best practices.